src/protocols/secure_channel/BUILD.gn

```diff
@@ -38,5 +39,18 @@ static_library("secure_channel") {
     "${chip_root}/src/system",
     "${chip_root}/src/tracing",
     "${chip_root}/src/tracing:macros",
+    "${chip_root}/src/transport",
+  ]
+
+  # secure channel requires messaging so it can send messages (e.g. for
```

Comment:
could we move Constants.h into a separate data_set and have both builds depend on it? It sounds like constants.h is reasonably stand-alone. Maybe this means adding Protocols.h and a few more, but still : constants can be in something like `:constants`

Reply:
> could we move Constants.h into a separate data_set and have both builds depend on it? 

We could for this specific thing, but in practice there are other secure channel bits (like status report) that should generally be usable from messaging and transport (and that one is what led me to find Constants.h not being listed).....  Fundamentally, the separation between messaging, transport and secure_channel as currently constituted is completely artificial; it's all the same thing in the spec.

Reply:
Sure, but it is code - we are supposed to try to layer them somehow.

Reply:
@andy31415 I mean, we could try to break apart "secure channel" into "secure channel basic data types" (Constants, status report, maybe some others) and having messaging/transport depend on those, and then having a "secure channel messaging" bit that depends on messaging/transport.  I can give that a shot.

Reply:
https://github.com/project-chip/connectedhomeip/pull/28248